### PR TITLE
Expose authdataresult.credential.secret in public header.

### PR DESCRIPTION
**Pull request by Yue-Wang-Google**

### CODE REVIEW

**Comment by Yue-Wang-Google** [comment]
Relevant design docs: https://docs.google.com/document/d/1iBprlRsVGcIPwodPD976N52gtEUKFFnZLH1YeVgz10o/edit# 
https://docs.google.com/document/d/1FZuAXbiMQK23aElPF7POms7Ae2zQpa8nKUx59Z52xFA/edit

File: Firebase/Auth/Source/Auth/FIRAuthDataResult_Internal.h [hunk 1]
```diff
@@ -44,7 +36,7 @@ NS_ASSUME_NONNULL_BEGIN
  */
 - (nullable instancetype)initWithUser:(FIRUser *)user
                    additionalUserInfo:(nullable FIRAdditionalUserInfo *)additionalUserInfo
-                           credential:(nullable FIROAuthCredential *)credential
+                           credential:(nullable FIRAuthCredential *)credential
```
Comment (renkelvin) [inline Firebase/Auth/Source/Auth/FIRAuthDataResult_Internal.h]: Why stop using a stricter type?

Comment (Yue-Wang-Google) [reply Firebase/Auth/Source/Auth/FIRAuthDataResult_Internal.h]: Done

**Comment by renkelvin** [comment]
LGTM. I'll approve and merge this pr when the launch is ready.

**Pull request merged**